File: tools/sms/src/main/java/top/bettercode/summer/tools/sms/SmsException.java
package top.bettercode.summer.tools.sms;

/**
 * @author Peter Wu
 */
public class SmsException extends RuntimeException {

  private static final long serialVersionUID = 1L;

  public SmsException() {
    this("短信平台请求失败");
  }

  public SmsException(String message) {
    super(message);
  }

  public SmsException(String message, Throwable cause) {
    super(message, cause);
  }

  public SmsException(Throwable cause) {
    this("短信平台请求失败", cause);
  }

  public SmsException(String message, Throwable cause, boolean enableSuppression,
      boolean writableStackTrace) {
    super(message, cause, enableSuppression, writableStackTrace);
  }
}

File: tools/sms/src/main/kotlin/top/bettercode/summer/tools/sms/SmsException.kt
package top.bettercode.summer.tools.sms

/**
 * @author Peter Wu
 */
class SmsException : RuntimeException {
    @JvmOverloads
    constructor(message: String? = "短信平台请求失败") : super(message)
    constructor(message: String?, cause: Throwable?) : super(message, cause)
    constructor(cause: Throwable?) : this("短信平台请求失败", cause)
    constructor(message: String?, cause: Throwable?, enableSuppression: Boolean,
                writableStackTrace: Boolean) : super(message, cause, enableSuppression, writableStackTrace)

    companion object
}
